### PR TITLE
[KRROOD] Added support for lists of enums in JSON and ORM.

### PR DESCRIPTION
**Pull request by tomsch420**
This pull request introduces several improvements and new features related to type serialization, enum handling, and forward reference resolution. The changes enhance the framework's ability to serialize/deserialize enums (including lists of enums), improve forward reference type resolution (especially for complex and iterative cases), and add utility functions for class inheritance introspection. There are also updates to the test suite to cover these new features and edge cases.

**Serialization and Enum Handling:**

* Added support for serializing and deserializing enums and lists of enums in the `JSONSerializableTypeRegistry`, including new `serialize_enum` and `deserialize_enum` functions, and registration of `enum.Enum` handlers. Also introduced `EnumListType` for handling lists of enums as JSON in the ORM layer. [[1]](diffhunk://#diff-5b88dd484952e653a0d0af346686f279dc32f474d9bee384fdcb6311b546fef0R3) [[2]](diffhunk://#diff-5b88dd484952e653a0d0af346686f279dc32f474d9bee384fdcb6311b546fef0R150-R153) [[3]](diffhunk://#diff-5b88dd484952e653a0d0af346686f279dc32f474d9bee384fdcb6311b546fef0R165-R168) [[4]](diffhunk://#diff-5b88dd484952e653a0d0af346686f279dc32f474d9bee384fdcb6311b546fef0L284-R331) [[5]](diffhunk://#diff-c7b5fcb53e566f3d4aa55f9e6e61e79d500ff9aa0564f86d13365ec8eb0c8b36R1-R10) [[6]](diffhunk://#diff-c7b5fcb53e566f3d4aa55f9e6e61e79d500ff9aa0564f86d13365ec8eb0c8b36R32-R62)
* Updated the deserializer function signatures to accept the target class as a parameter, improving flexibility and correctness, and updated UUID serialization/deserialization accordingly. [[1]](diffhunk://#diff-5b88dd484952e653a0d0af346686f279dc32f474d9bee384fdcb6311b546fef0L116-R117) [[2]](diffhunk://#diff-5b88dd484952e653a0d0af346686f279dc32f474d9bee384fdcb6311b546fef0L231-R240) [[3]](diffhunk://#diff-5b88dd484952e653a0d0af346686f279dc32f474d9bee384fdcb6311b546fef0L284-R331)

**Forward Reference and Type Resolution:**

* Refactored the forward reference resolution logic in `wrapped_field.py` to handle multiple unresolved types iteratively, building up the namespace until all references are resolved. This makes the type hint resolution more robust in complex scenarios.
* Added new test modules and classes to reproduce and verify forward reference resolution issues, including cases where referenced types are only available under `TYPE_CHECKING` or are not in `sys.modules`. [[1]](diffhunk://#diff-8548516cf43fc27b6e1ee7a9d38cb56aa3f159068034a0409e6a6315df6799e9R1-R68) [[2]](diffhunk://#diff-c164b746b9b7683bf7aea015e71055d6cc5d963b38fe6062479c672561686247R1-R51) [[3]](diffhunk://#diff-ceaea02173d7ebe43348aeeca210dc3b412f246472787bfa522eb454694788b9R1-R27)

**ORM and Utility Enhancements:**

* Added a utility function `inheritance_path_length` (with helper) to compute the minimum inheritance path between two classes, aiding in class introspection and ORM operations. [[1]](diffhunk://#diff-765bf3eac31d47b1115ad57d2363da41e280dbbc98a01b0b0239b61fc7183c3bR9) [[2]](diffhunk://#diff-765bf3eac31d47b1115ad57d2363da41e280dbbc98a01b0b0239b61fc7183c3bR208-R246)
* Updated `behaves_like_a_built_in_class` to recognize enums as built-in-like types, improving type handling in class diagram utilities.

**Test Coverage and Examples:**

* Added new example data classes and DAOs for enums, lists of enums, and forward reference types to the test dataset, ensuring the new features are properly tested and demonstrated. [[1]](diffhunk://#diff-69a8bc75b222afb2295f796a957b4e2d8b2cce6fc3a76ff1268b6b09a8fed182R607-R649) [[2]](diffhunk://#diff-05ee82abefc36cede7a5e6eb5f1557cb911d45622e6ee88a4645aec595b57cf3R727-R764) [[3]](diffhunk://#diff-05ee82abefc36cede7a5e6eb5f1557cb911d45622e6ee88a4645aec595b57cf3R840-R859)

**Error Messaging:**

* Improved error messages in `NoDAOFoundError` to provide clearer guidance when a DAO is missing, suggesting possible causes and remedies.

### CODE REVIEW

File: test/krrood_test/test_utils/test_json_serializer.py [hunk 1]
```diff
@@ -210,3 +216,10 @@ def test_with_kwargs():
     data = obj.to_json()
     result = from_json(data, b=2.0)
     assert obj == result
+
+
+def test_list_of_enums():
+    obj = [CustomEnum.A, CustomEnum.B]
+    data = to_json(obj)
+    result = from_json(data)
+    assert result == obj
```
Comment (ichumuh) [inline test/krrood_test/test_utils/test_json_serializer.py]: add tests for serializing exceptions

**Review by ichumuh — CHANGES_REQUESTED** [review]
i like the new json serializer

File: krrood/src/krrood/ormatic/custom_types.py [hunk 2]
```diff
@@ -25,3 +29,34 @@ def process_result_value(self, value: impl, dialect) -> Optional[Type]:
...
+        super().__init__()
+        self.enum_class = enum_class
+
+    def process_bind_param(self, values, dialect):
+        return {
+            JSON_TYPE_NAME: module_and_class_name(self.enum_class),
+            "values": [item.value for item in values],
+        }
+
+    def process_result_value(self, value: impl, dialect) -> Optional[List[enum.Enum]]:
```
Comment (AbdelrhmanBassiouny) [inline krrood/src/krrood/ormatic/custom_types.py]: doc

**Review by ichumuh — APPROVED** [review]

**Review by LucaKro — CHANGES_REQUESTED** [review]
